File: server_asp/src/models/Account.cs
﻿using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class Account
{
    public Guid Uid { get; set; }

    public string Username { get; set; } = null!;

    public string Password { get; set; } = null!;

    public string DisplayName { get; set; } = null!;

    public string Role { get; set; } = null!;

    public DateTime? ChangedPasswordAt { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual ICollection<Conversation> Conversations { get; set; } = new List<Conversation>();

    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: server_nodejs/src/models/account.js
// Account (UID, username, password, display_name, role)
module.exports = (sequelize, Sequelize) => {
    class Account extends Sequelize.Model {}

    Account.init(
        {
            uid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            username: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            password: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            display_name: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            role: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            changed_password_at: {
                type: Sequelize.DATE,
                allowNull: true,
            },
        },
        {
            sequelize,
            modelName: 'account',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    Account.associate = (models) => {
        Account.hasMany(models.conversation, {
            foreignKey: 'staffid',
            as: 'conversation',
        });
        Account.hasMany(models.order, {
            foreignKey: 'supported_by',
            as: 'order',
        });
    };
    return Account;
};
